lib/libzfs/libzfs_sendrecv.c

```diff
@@ -1152,66 +1152,79 @@ dump_snapshot(zfs_handle_t *zhp, void *arg)
 	return (err);
 }
 
+/*
+ * Send all snapshots for a filesystem, updating the send dump data.
+ */
 static int
-dump_filesystem(zfs_handle_t *zhp, void *arg)
+dump_filesystem(zfs_handle_t *zhp, send_dump_data_t *sdd)
 {
 	int rv = 0;
-	send_dump_data_t *sdd = arg;
 	boolean_t missingfrom = B_FALSE;
 	zfs_cmd_t zc = {"\0"};
 	uint64_t min_txg = 0, max_txg = 0;
 
+	/*
+	 * Make sure the tosnap exists.
+	 * XXX: This seems like it would be redundant?
+	 */
 	(void) snprintf(zc.zc_name, sizeof (zc.zc_name), "%s@%s",
 	    zhp->zfs_name, sdd->tosnap);
 	if (zfs_ioctl(zhp->zfs_hdl, ZFS_IOC_OBJSET_STATS, &zc) != 0) {
+		/* XXX: printing to stderr instead of logging to zhp? */
```

Comment:
Looks like some investigation is still needed for this commit at least.  In this case at least it looks like `stderr` was used since we're returning 0.